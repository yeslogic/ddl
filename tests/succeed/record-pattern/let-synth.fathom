let r = {x1 = false, x2 = false, x3 = {y1 = true, y2 = true}};
let {x1 = a1, x2 = a2, x3 = {y1 = b1, y2 = b2}} = r;
let {a, b} = {a = false, b = true};
let (x, _) = (false, true);
let (_, _) = (false, true);
{}
