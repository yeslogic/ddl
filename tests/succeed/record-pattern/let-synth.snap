stdout = '''
let r : { x1 : Bool, x2 : Bool, x3 : { y1 : Bool, y2 : Bool } } = {
    x1 = false,
    x2 = false,
    x3 = { y1 = true, y2 = true },
};
let a1 : Bool = r.x1;
let a2 : Bool = r.x2;
let b1 : Bool = r.x3.y1;
let b2 : Bool = r.x3.y2;
let _ : { a : Bool, b : Bool } = { a = false, b = true };
let a : Bool = _.a;
let b : Bool = _.b;
let _ : (Bool, Bool) = (false, true);
let x : Bool = _._0;
let _ : (Bool, Bool) = (false, true);
() : ()
'''
stderr = ''
